### PR TITLE
[HLS-3370]  - separate data-obfuscator package

### PR DESCRIPTION
**Pull request by LironKS**

### CODE REVIEW

File: sdk/sdk.go [hunk 1]
```diff
@@ -78,20 +79,23 @@ func WithMetadataOnlyMode() attribute.KeyValue {
 }
 
 func WithObfuscationBlocklistRules(blocklistRules []string) attribute.KeyValue {
+	os.Setenv(hsDataObfuscationBlocklistEnvVar, fmt.Sprint(blocklistRules))
```
Comment (nozik) [inline sdk/sdk.go:82]: Don't you need to `json.Marshall`?

File: data-obfuscator/go.mod [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module github.com/helios/go-sdk/dataobfuscator
+
+go 1.18
+
+require (
+	github.com/ohler55/ojg v1.17.4
+	go.opentelemetry.io/otel v1.12.0
```
Comment (nozik) [inline data-obfuscator/go.mod:7]: Why does the obfuscator rely on otel?

File: data-obfuscator/data_obfuscator_config.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	return HeliosObfuscationConfig{false, "", []jp.Expr{}, "0"}
+}
+
+func getObfuscationConfig() HeliosObfuscationConfig {
+	if reflect.ValueOf(obfuscationConfigSingelton).IsZero() {
+		obfuscationConfigSingelton = createObfuscationConfig()
+	}
+	return obfuscationConfigSingelton
+}
```
Comment (nozik) [inline data-obfuscator/data_obfuscator_config.go:71]: please run the linter

**Review by nozik — CHANGES_REQUESTED** [review]

File: sdk/sdk.go (hunk 1 above)
Comment (LironKS) [reply sdk/sdk.go:82]: we can, it's equivalent

File: data-obfuscator/go.mod (hunk 2 above)
Comment (LironKS) [reply data-obfuscator/go.mod:7]: it returns an attribute

Comment (nozik) [reply data-obfuscator/go.mod:7]: Which?

Comment (LironKS) [reply data-obfuscator/go.mod:7]: `func obfuscateAttributeValue(attribute attribute.KeyValue) attribute.Value {
	obfuscationConfig := getObfuscationConfig()
	hMacKey = []byte(obfuscationConfig.obfuscationhmacKey)
	if obfuscationConfig.obfuscationEnabled && slices.Contains(DATA_TO_OBFUSCATE, string(attribute.Key)) {
		return obfuscateDataHelper(attribute.Value, obfuscationConfig.obfuscationMode, obfuscationConfig.obfuscationRules)
	}
	return attribute.Value
}`

Comment (nozik) [reply data-obfuscator/go.mod:7]: 👍

**Review by nozik — APPROVED** [review]

**Pull request merged**